email/pardot_test.go

```diff
@@ -200,11 +203,39 @@ func TestSendContactRedactsEmail(t *testing.T) {
 	defer contactSrv.Close()
 
 	clk := clock.NewFake()
-	client, err := NewPardotClientImpl(clk, "biz-unit", "cid", "csec", tokenSrv.URL, contactSrv.URL)
+	client, err := NewPardotClientImpl(clk, "biz-unit", "cid", "csec", tokenSrv.URL, contactSrv.URL, nil)
 	test.AssertNotError(t, err, "failed to create client")
 
 	err = client.SendContact(emailToTest)
 	test.AssertError(t, err, "SendContact should fail")
 	test.AssertNotContains(t, err.Error(), emailToTest)
 	test.AssertContains(t, err.Error(), "[REDACTED]")
 }
+
+func TestSendContactDeduplication(t *testing.T) {
+	t.Parallel()
+
+	tokenSrv := httptest.NewServer(http.HandlerFunc(defaultTokenHandler))
+	defer tokenSrv.Close()
+
+	var contactHits int32
+	contactSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		atomic.AddInt32(&contactHits, 1)
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer contactSrv.Close()
+
+	cache := NewHashedEmailCache(1000, metrics.NoopRegisterer)
+	client, _ := NewPardotClientImpl(clock.New(), "biz", "cid", "csec",
+		tokenSrv.URL, contactSrv.URL, cache)
```

Comment:
We don't generally break function calls onto two lines like this, and none of the similar test functions above split it onto two lines.
```suggestion
	client, _ := NewPardotClientImpl(clock.New(), "biz", "cid", "csec", tokenSrv.URL, contactSrv.URL, cache)
```